app/models/user.rb

```diff
@@ -58,6 +58,14 @@ def self.from_oauth_hash(auth_hash)
     user
   end
 
+  def client_model
+    Proposal.client_model_for(self)
+  end
+
+  def client_model_slug
+    client_model.to_s.underscore.tr("/", "_")
```

Comment:
an example:
```
[3] pry(main)> u = User.for_email 'peter.karman@gsa.gov'
  User Load (27.7ms)  SELECT  "users".* FROM "users" WHERE "users"."email_address" = $1 LIMIT 1  [["email_address", "peter.karman@gsa.gov"]]
=> #<User:0x007ffcda5d7418
 id: 664,
 email_address: "peter.karman@gsa.gov",
 first_name: "Peter",
 last_name: "Karman - XFB",
 created_at: Mon, 14 Sep 2015 17:09:04 UTC +00:00,
 updated_at: Fri, 13 Nov 2015 17:29:05 UTC +00:00,
 client_slug: "ncr",
 active: true>
[4] pry(main)> u.client_slug
=> "ncr"
[5] pry(main)> u.client_model_slug
=> "ncr_work_order"
```

the point is that it is theoretically possible for there to be more than one model in a client namespace. The `client_slug` is just the toplevel namespace. We want the specific model name as a single token (according to the ES tokenization rules).